### ISSUE TITLE
[BUG] pyscenic grn - AttributeError: 'DataFrame' object has no attribute 'as_matrix'

### ISSUE THREAD

**Issue by c5creative** [aertslab/pySCENIC#242]
**Describe the bug**
Error when running pyscenic grn - `AttributeError: 'DataFrame' object has no attribute 'as_matrix'`

**Steps to reproduce the behavior**
1. Command run when the error occurred:

```python
pyscenic grn /path/to/file.loom /path/to/hs_hgnc_tfs.txt -o adj.csv --num_workers 12
```

2. Error encountered:

```pytb
2021-01-13 06:01:43,787 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.
2021-01-13 06:01:44,068 - pyscenic.cli.pyscenic - INFO - Inferring regulatory networks.
/data/private/conda/envs/pySCENIC/lib/python3.6/contextlib.py:88: UserWarning: Creating scratch directories is taking a surprisingly long time. This is often due to running workers on a network file system. Consider specifying a local-directory to point workers to write scratch data to a local disk.
  next(self.gen)
preparing dask client
parsing input
not shutting down client, client was created externally
finished
Traceback (most recent call last):
  File "/data/private/conda/envs/pySCENIC/bin/pyscenic", line 8, in <module>
    sys.exit(main())
  File "/data/private/conda/envs/pySCENIC/lib/python3.6/site-packages/pyscenic/cli/pyscenic.py", line 470, in main
    args.func(args)
  File "/data/private/conda/envs/pySCENIC/lib/python3.6/site-packages/pyscenic/cli/pyscenic.py", line 74, in find_adjacencies_command
    network = method(expression_data=ex_mtx, tf_names=tf_names, verbose=True, client_or_address=client, seed=args.seed)
  File "/data/private/conda/envs/pySCENIC/lib/python3.6/site-packages/arboreto/algo.py", line 41, in grnboost2
    early_stop_window_length=early_stop_window_length, limit=limit, seed=seed, verbose=verbose)
  File "/data/private/conda/envs/pySCENIC/lib/python3.6/site-packages/arboreto/algo.py", line 115, in diy
    expression_matrix, gene_names, tf_names = _prepare_input(expression_data, gene_names, tf_names)
  File "/data/private/conda/envs/pySCENIC/lib/python3.6/site-packages/arboreto/algo.py", line 214, in _prepare_input
    expression_matrix = expression_data.as_matrix()
  File "/data/private/conda/envs/pySCENIC/lib/python3.6/site-packages/pandas/core/generic.py", line 5139, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'as_matrix'
```

**Expected behavior**
 Completion of pyscenic grn and generation of results/adjacencies.csv

**Please complete the following information:**
- pySCENIC version: 0.10.4
- Installation method: Conda
- Run environment: Jupyter notebook **and** CLI
- OS: CentOS Linux release 7.6.1810 (Core)
- Package versions: [obtain using `pip freeze`, `conda list`, or skip this if using Docker/Singularity]:

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
anndata                   0.7.5                    pypi_0    pypi
arboreto                  0.1.5                    pypi_0    pypi
attrs                     20.3.0                   pypi_0    pypi
blosc                     1.20.1               h9c3ff4c_0    conda-forge
boltons                   20.2.1                   pypi_0    pypi
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.17.1               h36c2ea0_0    conda-forge
ca-certificates           2020.12.5            ha878542_0    conda-forge
cached-property           1.5.2                    pypi_0    pypi
cairo                     1.16.0            h9f066cc_1006    conda-forge
certifi                   2020.12.5        py36h5fab9bb_0    conda-forge
cffi                      1.14.4           py36h261ae71_0
click                     7.1.2                    pypi_0    pypi
cloudpickle               1.6.0                    pypi_0    pypi
cycler                    0.10.0                     py_2    conda-forge
cytoolz                   0.11.0           py36h7b6447c_0    anaconda
dask                      1.0.0                    pypi_0    pypi
decorator                 4.4.2                    pypi_0    pypi
dill                      0.3.3                    pypi_0    pypi
distributed               1.28.1                   pypi_0    pypi
expat                     2.2.9                he1b5a44_2    conda-forge
fontconfig                2.13.1            h7e3eb15_1002    conda-forge
freetype                  2.10.4               h7ca028e_0    conda-forge
frozendict                1.2                      pypi_0    pypi
gdbm                      1.18                 h0a1914f_2    conda-forge
get-version               2.1                      pypi_0    pypi
gettext                   0.19.8.1          h0b5b191_1005    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
h5py                      3.1.0                    pypi_0    pypi
hdf5                      1.10.5          nompi_h5b725eb_1112    conda-forge
heapdict                  1.0.1                    pypi_0    pypi
icu                       67.1                 he1b5a44_0    conda-forge
igraph                    0.8.3                hef4adab_1    conda-forge
importlib-metadata        3.1.1                    pypi_0    pypi
interlap                  0.2.7                    pypi_0    pypi
joblib                    0.17.0                     py_0    conda-forge
jpeg                      9d                   h36c2ea0_0    conda-forge
kiwisolver                1.3.1            py36h51d7077_0    conda-forge
krb5                      1.17.2               h926e7f8_0    conda-forge
lcms2                     2.11                 hcbb858e_1    conda-forge
ld_impl_linux-64          2.35.1               hed1e6ac_0    conda-forge
legacy-api-wrap           1.2                      pypi_0    pypi
libblas                   3.9.0                3_openblas    conda-forge
libcblas                  3.9.0                3_openblas    conda-forge
libcurl                   7.71.1               hcdd3856_8    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.3                  h58526e2_2    conda-forge
libgcc-ng                 9.3.0               h5dbcf3e_17    conda-forge
libgfortran-ng            9.3.0               he4bcb1c_17    conda-forge
libgfortran4              7.5.0               hae1eefd_17    conda-forge
libgfortran5              9.3.0               he4bcb1c_17    conda-forge
libglib                   2.66.3               h1f3bc88_1    conda-forge
libgomp                   9.3.0               h5dbcf3e_17    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
liblapack                 3.9.0                3_openblas    conda-forge
libllvm10                 10.0.1               he513fc3_3    conda-forge
libllvm8                  8.0.1                hc9558a2_0    conda-forge
libnghttp2                1.41.0               h8cfc5f6_2    conda-forge
libopenblas               0.3.12          pthreads_h4812303_1    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libssh2                   1.9.0                hab1572f_5    conda-forge
libstdcxx-ng              9.3.0               h2ae2ef3_17    conda-forge
libtiff                   4.1.0                h4f3a223_6    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libwebp-base              1.1.0                h36c2ea0_3    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.10               h68273f3_2    conda-forge
llvmlite                  0.31.0           py36hfa65bc7_1    conda-forge
loompy                    3.0.6                    pypi_0    pypi
louvain                   0.6.1            py36h831f99a_3    conda-forge
lz4-c                     1.9.2                he1b5a44_3    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
matplotlib                3.3.2                         0    conda-forge
matplotlib-base           3.3.2            py36he12231b_1    conda-forge
mock                      4.0.2            py36h9f0ad1d_1    conda-forge
msgpack                   1.0.0                    pypi_0    pypi
multicore-tsne            0.1_d4ff4aab     py36h3e44d54_0    conda-forge
multiprocessing-on-dill   3.5.0a4                  pypi_0    pypi
natsort                   7.1.0                    pypi_0    pypi
ncurses                   6.2                  h58526e2_4    conda-forge
networkx                  2.5                      pypi_0    pypi
numba                     0.48.0           py36hb3f55d8_0    conda-forge
numexpr                   2.7.1            py36h7c3b610_3    conda-forge
numpy                     1.19.4           py36h8732dcd_1    conda-forge
numpy-groupies            0.9.13                   pypi_0    pypi
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openssl                   1.1.1h               h516909a_0    conda-forge
packaging                 20.7                     pypi_0    pypi
pandas                    0.25.3                   pypi_0    pypi
patsy                     0.5.1                      py_0    conda-forge
pcre                      8.44                 he1b5a44_0    conda-forge
pillow                    8.0.1            py36h10ecd5c_0    conda-forge
pip                       20.3.1             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               h36c2ea0_0    conda-forge
psutil                    5.7.3                    pypi_0    pypi
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pyarrow                   0.16.0                   pypi_0    pypi
pycairo                   1.20.0           py36h4779a57_1    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pypy3.6                   7.3.3                he1ebe50_2    conda-forge
pyscenic                  0.10.4                   pypi_0    pypi
pytables                  3.6.1            py36h9f153d1_1    conda-forge
python                    3.6.11          hffdb5ce_3_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-igraph             0.8.3            py36h76a8ce2_2    conda-forge
python_abi                3.6                     1_cp36m    conda-forge
pytz                      2020.4             pyhd8ed1ab_0    conda-forge
pyyaml                    5.3.1                    pypi_0    pypi
readline                  8.0                  he28a2e2_2    conda-forge
scanpy                    1.6.0                    pypi_0    pypi
scikit-learn              0.23.2           py36hb6e6923_3    conda-forge
scipy                     1.5.3            py36h9e8f40b_0    conda-forge
seaborn                   0.11.0               ha770c72_1    conda-forge
seaborn-base              0.11.0             pyhd8ed1ab_1    conda-forge
setuptools                49.6.0           py36h9880bd3_2    conda-forge
setuptools-scm            4.1.2                    pypi_0    pypi
sinfo                     0.3.1                    pypi_0    pypi
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sortedcontainers          2.3.0                    pypi_0    pypi
sqlite                    3.34.0               h74cdb3f_0    conda-forge
statsmodels               0.11.1           py36h516909a_0    conda-forge
stdlib-list               0.8.0                    pypi_0    pypi
tblib                     1.7.0                    pypi_0    pypi
texttable                 1.6.3              pyh9f0ad1d_0    conda-forge
threadpoolctl             2.1.0              pyh5ca1d4c_0    conda-forge
tk                        8.6.10               hed695b0_1    conda-forge
toolz                     0.11.1                     py_0    anaconda
tornado                   6.1              py36h1d69622_0    conda-forge
tqdm                      4.54.1                   pypi_0    pypi
umap-learn                0.4.6                    pypi_0    pypi
wheel                     0.36.1             pyhd3deb0d_0    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.12               h516909a_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
zict                      2.0.0                    pypi_0    pypi
zipp                      3.4.0                    pypi_0    pypi
zlib                      1.2.11            h516909a_1010    conda-forge
zstd                      1.4.5                h6597ccf_2    conda-forge
```

**Comment by cflerin** [comment]
Hi @c5creative , thanks for the bug report. Working on fixing this, but for now, you can do one of two things: downgrade pandas to <1.0.0, or use the pyscenic dev branch.

**Comment by cflerin** [comment]
This is fixed in the `0.11.0` release of pySCENIC (Arboreto `0.1.6`)
